accepted/0000-caasp-integration.md

```diff
@@ -0,0 +1,298 @@
+
+  [`skuba-update` will take care of updating the Base Operating System](https://susedoc.github.io/doc-caasp/beta/caasp-admin/single-html/#_base_os): this means we cannot register any master or worker node with SUSE Manager (otherwise an end-user could apply updates to the machine).
+  In case the machines are registered to SUSE Manager, the machines will be de-registered during the bootstrapping.
+  
+  ###### Bootstrapping master and worker nodes
+
+  In SUSE Manager under "CaaS Platform > Deployment > cluster name" there will be a list of machines that:
+  - are not part of any other CaaS Platform cluster
+  - are not already bootstrapped into the current cluster
+  - satisfy the CaaS Platform node requirements: those machines are either bootstrapped via the AutoYaST profile or are already registered with SUSE Manager and satisfy the node requirements (checked via grains).
+
+  The user will then have the option to bootstrap the selected machine as the first master (or subsequently, as an additional master or worker). Upon triggering of this event, the SUSE Manager will:
+
+  - operate on the to-be master or worker node:
+    - Copy the SSH key of the Management Node for passwordless login in the node. In case of existing machines, the SSH file must be copied (e.g. via Salt), whereas the AutoYaST case will be already covered during provisioning.
+    - Assign the CaaS Platform channels assigned via Salt (SUSE CaaS Platform Extension to SUSE Linux Enterprise 15 and SUSE Containers Module 15 SP1 x86_64), whereas the AutoYaST case will be already covered during provisioning.
+    - De-register the machine from SUSE Manager (if it is registered)
+
+  - operate on the Management Node:
+    - Salt will invoke `skuba` (using `cmd.run`) to bootstrap the first master or make the selected node join the cluster. The user must specify which user to use for passwordless login (in this case `sles`) and whether to use `sudo` or not.
+
+    Example:
+
+    ```
+    bootstrap_first_master:
+      cmd.run:
+        - name: skuba node bootstrap --user sles --sudo --target <node IP> <node name>
+    
+    bootstrap_additional_master:
+      cmd.run:
+        - name: skuba node join --role master --user sles --sudo --target <node IP> <node name>
+    
+    bootstrap_worker:
+      cmd.run:
+        - name: skuba node join --role worker --user sles --sudo --target <node IP> <node name>
+    ```
+
+    The output of the `skuba` command must be parsed and shown to the user in SUSE Manager in case of errors.
+
+## Manage a CaaS Platform cluster using SUSE Manager
+
+### Show cluster status
+
+SUSE Manager shows the status of the cluster by invoking `skuba cluster status` on the Management Node (via Salt).
```

Comment:
There's no need to run this command from the management node, as long as `skuba` has an `admin.conf` `kubeconfig` file it can use to login into the cluster.

Reply:
In the SUSE Manager ecosystem, only the `Management Node` will have `skuba` deployed and the set of all the nodes SSH keys deployed.